src/module_2/EDA_1.md

```diff
@@ -0,0 +1,3314 @@
+    mean   3.694880e+13     6.307351          7.028881
+    std    2.725674e+12     7.107218          7.660542
+    min    3.361529e+13     0.000000          0.000000
+    25%    3.427657e+13     2.490000          2.850000
+    50%    3.927260e+13     3.990000          4.490000
+    75%    3.948318e+13     7.490000          8.210000
+    max    4.016793e+13    59.990000         60.000000
+           count_people  count_adults  count_children  count_babies  count_pets
+    count    325.000000    325.000000      325.000000    325.000000  325.000000
+    mean       2.787692      2.003077        0.707692      0.076923    0.636923
+    std        1.365753      0.869577        1.026246      0.289086    0.995603
+    min        0.000000      0.000000        0.000000      0.000000    0.000000
+    25%        2.000000      2.000000        0.000000      0.000000    0.000000
+    50%        3.000000      2.000000        0.000000      0.000000    0.000000
+    75%        4.000000      2.000000        1.000000      0.000000    1.000000
+    max        8.000000      7.000000        6.000000      2.000000    6.000000
+
+
+
+```python
+#OrdersAnalysis
+
+
+import matplotlib.pyplot as plt
+#Missingvalues
+#msno.bar(orders)
+
+# Número total de órdenes
+num_orders = orders.shape[0]
+print(f"Número total de órdenes: {num_orders}")
+
+# Calcular el número de ítems por pedido
+orders['num_items'] = orders['ordered_items'].apply(len)
+
+# Promedio de ítems por pedido
+avg_items_per_order = orders['num_items'].mean()
+print(f"Promedio de ítems por pedido: {avg_items_per_order:.2f}")
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+# Histograma de la distribución de ítems por pedido
+sns.histplot(orders['ordered_items'], bins=20)
+plt.title('Distribución de Ítems por Pedido')
+plt.xlabel('Número de Ítems')
+plt.ylabel('Frecuencia')
+plt.show()
+
+# Asegúrate de que la columna de fecha esté en formato datetime
+orders['order_date'] = pd.to_datetime(orders['order_date'])
+
+# Agrupar por mes y contar las órdenes
+orders_per_month = orders.groupby(orders['order_date'].dt.to_period('M')).size()
+
+# Visualizar la tendencia de órdenes a lo largo del tiempo
+orders_per_month.plot(kind='line', marker='o', figsize=(10,6))
+plt.title('Tendencia de Órdenes a lo Largo del Tiempo')
+plt.xlabel('Fecha')
+plt.ylabel('Número de Órdenes')
+plt.show()
+
+from collections import Counter
+
+# Aplanar la lista de ítems comprados
+all_items = [item for sublist in orders['ordered_items'] for item in sublist]
+
+# Contar la frecuencia de cada ítem
+item_counts = Counter(all_items)
+
+# Convertir a un DataFrame y mostrar los productos más comprados
+item_counts_df = pd.DataFrame(item_counts.items(), columns=['item_id', 'count']).sort_values(by='count', ascending=False)
+
+print("Productos más comprados:")
+print(item_counts_df.head())
+
+# Visualización de los productos más comprados
+top_items = item_counts_df.head(10)
+sns.barplot(x='count', y='item_id', data=top_items)
```

Comment:
This plots is of no help. A table is much more informative. Also the product ids are wrong: they cannot be a single letter